trie/stacktrie.go

```diff
@@ -43,13 +50,32 @@ func (o *StackTrieOptions) WithWriter(writer func(path []byte, hash common.Hash,
 	return o
 }
 
+// WithCleaner configures the cleaner in the option for removing dangling nodes.
+func (o *StackTrieOptions) WithCleaner(cleaner func(path []byte)) *StackTrieOptions {
+	o.Cleaner = cleaner
+	return o
+}
+
+// SkipBoundary configures whether the left and right boundary nodes are filtered
+// for committing, along with a onBoundary callback invoked whenever the boundary
+// nodes are met.
+func (o *StackTrieOptions) SkipBoundary(skipLeft, skipRight bool, gauge metrics.Gauge) *StackTrieOptions {
```

Comment:
If we're going with the functional options approach, the convention is to prefix the modified with `With`, otherwise it's confusing as to what the difference it (i.e. I needed to look it up before I was convinced it's not a bug).

In that vein, pls rename to `WithSkipBoundary`.